llvm/test/tools/sycl-post-link/composite-spec-constant.ll

```diff
@@ -0,0 +1,91 @@
+; RUN: sycl-post-link -spec-const=rt --ir-output-only %s -S -o - \
+; RUN: | FileCheck %s --implicit-check-not __sycl_getCompositeSpecConstantValue
+;
+; This test is intended to check that sycl-post-link tool is capable of handling
+; composite specialization constants by lowering them into a set of SPIR-V
+; friendly IR operations representing those constants.
+;
+; CHECK: %[[#NS0:]] = call i32 @_Z20__spirv_SpecConstantii(i32 [[#ID:]], i32
+; CHECK: %[[#NS1:]] = call float @_Z20__spirv_SpecConstantif(i32 [[#ID + 1]], float
+; CHECK: %[[#NA0:]] = call %struct._ZTS1A.A @_Z29__spirv_SpecConstantCompositeif(i32 %[[#NS0]], float %[[#NS1]])
+;
+; CHECK: %[[#NS2:]] = call i32 @_Z20__spirv_SpecConstantii(i32 [[#ID + 3]], i32
+; CHECK: %[[#NS3:]] = call float @_Z20__spirv_SpecConstantif(i32 [[#ID + 4]], float
+; CHECK: %[[#NA1:]] = call %struct._ZTS1A.A @_Z29__spirv_SpecConstantCompositeif(i32 %[[#NS2]], float %[[#NS3]])
+;
+; CHECK: %[[#NA:]] = call [2 x %struct._ZTS1A.A] @_Z29__spirv_SpecConstantCompositestruct._ZTS1A.Astruct._ZTS1A.A(%struct._ZTS1A.A %[[#NA0]], %struct._ZTS1A.A %[[#NA1]])
+;
+; CHECK: %[[#B0:]] = call i32 @_Z20__spirv_SpecConstantii(i32 [[#ID + 7]], i32{{.*}})
+; CHECK: %[[#B1:]] = call i32 @_Z20__spirv_SpecConstantii(i32 [[#ID + 8]], i32{{.*}})
```

Comment:
Why `ID + 2`, `ID + 5`, `ID + 6` are missing? Can we make ids consecutive?

Reply:
> Why `ID + 2`, `ID + 5`, `ID + 6` are missing? 

I allocate a separate ID for each composite encountered

> Can we make ids consecutive?

Theoretically, yes, because composites don't have IDs in SPIR-V

Reply:
I made IDs consecutive in 9a82d5320c6a9e228fc183f211980b8de5d58d37